models/issue_user.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2017 The Gogs Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"fmt"
+)
+
+// IssueUser represents an issue-user relation.
+type IssueUser struct {
+	ID          int64 `xorm:"pk autoincr"`
+	UID         int64 `xorm:"INDEX"` // User ID.
+	IssueID     int64
+	IsRead      bool
+	IsAssigned  bool
+	IsMentioned bool
+}
+
+func newIssueUsers(e Engine, repo *Repository, issue *Issue) error {
```

Comment:
Changed `e` to `Engine` (was previously `*xorm.Session`). Similarly for `updateIssueUserByAssignee(..)`